### ISSUE TITLE
Allow developers to change the toggle button icon

### ISSUE THREAD

**Issue by jouni** [vaadin/vaadin-combo-box#196]
As a workaround for situations like #191

Consider the use case of hiding the toggle button completely at the same time.

**Comment by ktiedt** [comment]
A common use case for this style of element is searching through a list, in the MD spec there would be a 'search' icon (magnifier) on the left and no toggle on the right

Just to elaborate on possible uses here :)

**Comment by jounik** [comment]
Slightly related to #198, which would allow you to add the search icon as a prefix element.

**Comment by thockaday1** [comment]
Just wanted to share a (pretty unrefined) workaround I've been using to target the icon in case it's useful for anyone needing to do something along these lines. 

Example setting the icon to 'arrow-drop-down-circle' for a vaadin-combo-box with an id of 'myComboBox':
this.$.myComboBox.querySelectorAll('iron-icon')[1].icon = 'arrow-drop-down-circle';

Hope this helps. This method worked for what I needed at the time but I've switched to the newer version with the arrow-drop-down - the code would need adjustment if another iron-icon was added before the dropdown icon, but the concept would still be similar.

**Comment by platosha** [comment]
@thockaday1 hi! Thanks for sharing your workaround.

I’d like to warn you and everyone: __this workaround works only when Polymer is in Shady DOM mode (current default)__. If you switch it to Shadow DOM, then local DOM contents would be hidden from querying outside of the vaadin-combo-box, thus `.querySelectorAll('iron-icon')` would be empty.

In order to try your ideas with Shadow DOM mode, you can temporarily switch Polymer mode to use Shadow DOM by adding `?dom=shadow` query string to your page URL.

Shadow DOM is still not supported natively in other browsers except Chrome, but still could be enabled in most of the browsers if you include the full webcomponents.js polyfill (not the -lite version).

In general, __modifying local DOM contents of the vaadin-combo-box and other Polymer elements is a bad practice. Avoid if possible.__ Never do that if you are not in control of the entire application.

**Comment by thockaday1** [comment]
Well put @platosha - thanks for the clarification. Like I said, it worked for what I needed at the time. I'm using the standard one now that the icon is the arrow-drop-down - thanks for that update!

**Comment by thockaday1** [comment]
@platosha - I did some more digging on the local DOM API and playing around with what I had to alter the method sightly, and I was able to get the icon to render properly in Shadow DOM. Here's what I ended up with:
```
Polymer.dom(this.$.myComboBox.root).querySelectorAll('iron-icon')[1].icon = 'arrow-drop-down-circle';
```

Before changing the code, I just had 
```
this.$.myComboBox.querySelectorAll('iron-icon')[1].icon = 'arrow-drop-down-circle';
```

Any thoughts on this approach or drawbacks to be aware of? Thanks again for the feedback!

**Comment by platosha** [comment]
@ghost this is better, now that it works both for Shady and Shadow DOM modes. But of course, this line still depends on the internals of vaadin-combo-box, which are not intended to be modified like this. There’s no guarantee of future versions compatibility, so be careful: this might get broken after some update.

**Comment by Saulis** [comment]
Fixed in #254
